src/fswAlgorithms/effectorInterfaces/solarArrayReference/solarArrayReference.c

```diff
@@ -69,5 +71,98 @@ void Reset_solarArrayReference(solarArrayReferenceConfig *configData, uint64_t c
 */
 void Update_solarArrayReference(solarArrayReferenceConfig *configData, uint64_t callTime, int64_t moduleID)
 {
-    
+     /*! - Create and assign buffer messages */
+    NavAttMsgPayload            attNavIn = NavAttMsg_C_read(&configData->attNavInMsg);
+    AttRefMsgPayload            attRefIn = AttRefMsg_C_read(&configData->attRefInMsg);
+    HingedRigidBodyMsgPayload   hingedRigidBodyIn     = HingedRigidBodyMsg_C_read(&configData->hingedRigidBodyInMsg);
+    HingedRigidBodyMsgPayload   hingedRigidBodyRefOut = HingedRigidBodyMsg_C_zeroMsgPayload();
+
+    /*! read Sun direction in B frame from the attNav message and map it to R frame */
+    double rHat_SB_B[3];    // Sun direction in body-frame coordinates
+    double rHat_SB_R[3];    // Sun direction in reference-frame coordinates
+    double BN[3][3];   // inertial to body frame DCM
+    double RN[3][3];   // inertial to reference frame DCM
+    double RB[3][3];   // body to reference DCM
+    v3Normalize(attNavIn.vehSunPntBdy, rHat_SB_B);
+    switch (configData->bodyFrame) {
+
+        case 0:
+        MRP2C(attNavIn.sigma_BN, BN);
+        MRP2C(attRefIn.sigma_RN, RN);
+        m33MultM33t(RN, BN, RB);
+        m33MultV3(RB, rHat_SB_B, rHat_SB_R);
+        break;
+
+        case 1:
+        v3Copy(rHat_SB_B, rHat_SB_R);
+        break;
+
+        default:
+        _bskLog(configData->bskLogger, BSK_ERROR, "Error: solarArrayAngle.bodyFrame input can be either 0 or 1.");
```

Comment:
:large_orange_diamond: Same thing with "Error: " string

Reply:
Fixed it.